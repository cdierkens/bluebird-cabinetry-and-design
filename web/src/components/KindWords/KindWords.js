import { graphql, useStaticQuery } from "gatsby";
import PropTypes from "prop-types";
import React, { useState } from "react";
import "react-responsive-carousel/lib/styles/carousel.min.css";
import Button from "../Button";
import Container from "../container";

const query = graphql`
  query KindWordsQuery {
    allSanityClientReview(sort: { fields: [_createdAt] }) {
      nodes {
        _id
        clientName
        text
        title
      }
    }
  }
`;

const KindWordsItem = ({ title, children, name }) => (
  <div className="leading-relaxed text-center px-2 max-w-screen-md mx-auto">
    <div className="relative w-full h-full flex justify-center items-center flex-col border border-gray-light rounded-md px-3 py-3 pt-6  mb-12 shadow-md">
      {title && (
        <span className="leading-none font-sans text-3xl bg-white absolute top-0 inset-x-auto transform -translate-y-1/2 px-4">
          &ldquo;{title}&rdquo;
        </span>
      )}
      <p className="m-0 mb-1">{children}</p>
      <p className="m-0 italic text-gray-darker">-{name}</p>
    </div>
  </div>
);

KindWordsItem.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node,
  name: PropTypes.string,
};

const KindWords = () => {
  const [showReadMore, setShowMore] = useState(true);
  const {
    allSanityClientReview: { nodes: kindWords },
  } = useStaticQuery(query);

  return (
    <Container className="py-6">
      <h2 className="text-center">Kind Words</h2>

      {kindWords.slice(0, 3).map(({ title, clientName, text, _id }) => (
        <KindWordsItem title={title} name={clientName} key={_id}>
          {text}
        </KindWordsItem>
      ))}

      {!showReadMore &&
        kindWords.slice(3).map(({ title, clientName, text, _id }) => (
          <KindWordsItem title={title} name={clientName} key={_id}>
            {text}
          </KindWordsItem>
        ))}

      {showReadMore && (
        <div className="text-center">
          <Button
            type="button"
            variant="blue"
            onClick={() => setShowMore(false)}
          >
            Read More
          </Button>
        </div>
      )}
    </Container>
  );
};

export default KindWords;
