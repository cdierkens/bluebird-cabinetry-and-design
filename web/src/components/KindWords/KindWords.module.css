.KindWordsContainer {
  @apply py-8 grid grid-cols-8;
}

.KindWordsH2 {
  @apply text-center text-blue-dark col-span-8;
}

.Button {
  @apply font-body font-medium rounded-md border border-blue-dark text-blue-dark py-3 px-9 my-4 mx-4 text-xl col-span-2 text-center col-start-4;
  @apply transition-colors duration-500 ease-in-out bg-transparent;
}

.Button:hover,
.Button:active,
.Button:focus {
  @apply bg-white;
}
