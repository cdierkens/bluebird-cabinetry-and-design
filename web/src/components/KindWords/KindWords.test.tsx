import { render } from "@testing-library/react";
import React from "react";
import KindWords from "./KindWords";

it("matches snapshot", () => {
  const { asFragment } = render(<KindWords />);

  expect(asFragment()).toMatchSnapshot();
});
