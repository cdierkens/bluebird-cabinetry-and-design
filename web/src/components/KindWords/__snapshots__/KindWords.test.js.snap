// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Container py-6 "
  >
    <h2
      class="text-center"
    >
      Kind Words
    </h2>
    <div
      class="text-center"
    >
      <button
        class="ButtonBlue  focus:outline-none focus:ring"
        type="button"
      >
        Read More
      </button>
    </div>
  </div>
</DocumentFragment>
`;
